### PR TITLE
Add process html attributes to the PathProcessor interface

### LINKED ISSUE #1702: Populate htmlAttributes of Links globally on a project
As a developer I want to be able to modify htmlAttributes of Links globally on a project level.

Acceptance Criteria:
In a project implementation code the htmlAttributes of Links can be modified based on the provided htmlAttributes and the linkUrl.

### PR DESCRIPTION
**Pull request by bpauli**
- extend PathProcessor interface to process html attributes
- add default handling to the DefaultPathProcessor

| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #1702 
| Patch: Bug Fix?          |
| Minor: New Feature?      |
| Major: Breaking Change?  |
| Tests Added + Pass?      | Yes
| Documentation Provided   | Yes (code comments and or markdown)
| Any Dependency Changes?  |
| License                  | Apache License, Version 2.0

### CODE REVIEW

**Review by comanV — APPROVED** [review]

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/services/link/PathProcessor.java [hunk 1]
```diff
@@ -64,4 +67,13 @@
      * @return the external link of the given path
      */
     @NotNull String externalize(@NotNull String path, @NotNull SlingHttpServletRequest request);
+
+
+    /**
+     * Process the HTML attributes for the {@link com.adobe.cq.wcm.core.components.internal.link.LinkHandler}
```
Comment (jckautzmann) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/services/link/PathProcessor.java:73]: I would rephrase it as follows:
Processes HTML attributes and returns the processed attributes.
This is an extension point to modify HTML attributes, remove or add new ones.

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/services/link/PathProcessor.java [hunk 2]
```diff
@@ -64,4 +67,13 @@
...
+     * @param path the path of the linked resource
```
Comment (jckautzmann) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/services/link/PathProcessor.java:74]: Is the path needed to process HTML attributes?

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/services/link/PathProcessor.java [hunk 3]
```diff
@@ -64,4 +67,13 @@
...
+     * @param htmlAttributes the origin HTML attributes of the link
```
Comment (jckautzmann) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/services/link/PathProcessor.java:75]: I'd rephrase it as: the HTML attributes to be processed as they can be any attributes, not just the original ones.
